### ISSUE TITLE
Error when deleting objects from database in dev branch

### ISSUE THREAD

**Issue by crbaird** [warewulf/warewulf3#85]
Attempting to remove a bootstrap file from the database results in the following error:

DBD::mysql::st execute failed: Cannot delete or update a parent row: a foreign key constraint fails (`warewulf`.`binstore`, CONSTRAINT `binstore_ibfk_1` FOREIGN KEY (`object_id`) REFERENCES `datastore` (`id`)) at /usr/lib/perl5/vendor_perl/5.18.2/Warewulf/DataStore/SQL/BaseClass.pm line 1121, <FIN> line 1.
WARNING:  Unable to execute datastore removal query: Cannot delete or update a parent row: a foreign key constraint fails (`warewulf`.`binstore`, CONSTRAINT `binstore_ibfk_1` FOREIGN KEY (`object_id`) REFERENCES `datastore` (`id`))
Deleting local bootable bootstrap files: 4.4.21-69-default

It looks like the binstore scrubber isn't being passsed the correct object_id.

**Comment by bensallen** [comment]
@jtfrey Any chance you could take a look at this? I've seen this pop up at well.

**Comment by jtfrey** [comment]
At lib/Warewulf/DataStore/SQL/BaseClass.pm:1099 in the development branch:

~~~~
    #
    # Scrub the binstore:
    #
    if ( $self->{"BINSTORE_KIND"} eq $BINSTORE_KIND_DATABASE && ! $self->del_object_binstore_db_impl($buffer) ) {
        goto EARLY_EXIT;
    }
    elsif ( $self->{"BINSTORE_KIND"} eq $BINSTORE_KIND_FILESYSTEM && ! $self->del_object_binstore_fs_impl($buffer) ) {
        goto EARLY_EXIT;
    }
~~~~

The `*_impl` functions should be called with `$object_id` as the argument and not `$buffer`.

**Comment by jtfrey** [comment]
I've patched the code myself, pull request #92 is in.

**Comment by bensallen** [comment]
Thanks @jtfrey. Fixed in #92.
